File: software/examples/c/example_simple.c
#include <stdio.h>

#include "ip_connection.h"
#include "brick_imu.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XXYYZZ" // Change XXYYZZ to the UID of your IMU Brick

int main(void) {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	IMU imu;
	imu_create(&imu, UID, &ipcon);

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		return 1;
	}
	// Don't use device before ipcon is connected

	// Get current quaternion
	float x, y, z, w;
	if(imu_get_quaternion(&imu, &x, &y, &z, &w) < 0) {
		fprintf(stderr, "Could not get quaternion, probably timeout\n");
		return 1;
	}

	printf("Quaternion [X]: %f\n", x);
	printf("Quaternion [Y]: %f\n", y);
	printf("Quaternion [Z]: %f\n", z);
	printf("Quaternion [W]: %f\n", w);

	printf("Press key to exit\n");
	getchar();
	imu_destroy(&imu);
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
	return 0;
}

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"tinkerforge/imu_brick"
	"tinkerforge/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XXYYZZ" // Change XXYYZZ to the UID of your IMU Brick.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	imu, _ := imu_brick.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current quaternion.
	x, y, z, w, _ := imu.GetQuaternion()

	fmt.Printf("Quaternion [X]: %f\n", x)
	fmt.Printf("Quaternion [Y]: %f\n", y)
	fmt.Printf("Quaternion [Z]: %f\n", z)
	fmt.Printf("Quaternion [W]: %f\n", w)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
